src/cli.c

```diff
@@ -0,0 +1,80 @@
+#include "cli.h"
+#include "debug.h"
+
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <limits.h>
+#include <getopt.h>
+#include <stdio.h>
+
+#define MAX_HZ (1000000L)
+
+static const char* usage_str = "cretro [-d <0..5>] <HERTZ> <ROM>";
+
+__attribute__((__const__)) config_t cli_config_default(void) {
+    config_t conf = {
+        .debug = 0,
+        .us_delay = -1,
+        .rom = "",
+    };
+
+    return conf;
+}
+
+int cli_config_handle(config_t* const conf, int argc, char **argv) {
+    int c;
+
+    // parse all options first
+    while ((c = getopt(argc, argv, "d:")) != -1) {
+        switch (c) {
+            case 'd':
+                conf->debug = (int) strtol(optarg, NULL, 10);
+                debug_init(conf);
+                break;
+            default:
+                fprintf(stderr, "%s\n", usage_str);
+                return EXIT_FAILURE;
+        }
+    }
+
+    // parse the remaining options: <HERTZ> <ROM>
+
+    // calculate delay from hertz input
+    long hertz = strtol(argv[optind++], NULL, 10);
+    LOG_DEBUG("Delay [Hz] input specified: %ld.", hertz);
+
+    if (hertz > MAX_HZ) {
+        LOG_FATAL("Provided delay %ld is higher than allowed delay %ld", hertz, MAX_HZ);
+        return EXIT_FAILURE;
+    }
+    else if (hertz <= 0) {
+        LOG_FATAL("Provided delay %ld is lower or equal to 0", hertz);
+        return EXIT_FAILURE;
+    }
+
+    // delay
+    if (optind >= argc) {
```

Comment:
This whole if-statement has to be moved up, as the input has already been processed for the frequency and the according delay
Additionally, the check has to be corrected itself, as the frequency-input is not the last one

Reply:
`optind` just gives the index of the currently reviewed item in `argv`, after `getopt` was used. This just checks, if enough arguments were passed to the application. Would this be omitted, an input like 
```
cretro -d 2 roms/bat.ch8
```
would lead to a crash.

Reply:
But I think that he has a point with regards to the count 🤔 You could immediately print the usage string and return once it's clear that argc < 2 (the number of required parameters) without trying to parse.

Reply:
Yes that is true, but that is a different kind of check. I will implement it nonetheless.